python/ray/data/dataset.py

```diff
@@ -4599,7 +4599,9 @@ def stats(self) -> str:
             * Tasks per node: 20 min, 20 max, 20 mean; 1 nodes used
 
         """
-        if self._write_ds is not None and self._write_ds._plan.has_computed_output():
+        if self._current_executor:
```

Comment:
@raulchen Thoughts on adding something like this to pull stats from the streaming executor? This resolves the problems with mismatches between operator time and dataset time leading to more consistent throughput.

Reply:
looks good. makes sense